src/RootSolvers.jl

```diff
@@ -491,45 +1041,10 @@ function find_zero(
     tol::AbstractTolerance,
     maxiters::Int,
 ) where {F <: Function, FT}
-    local y
-    x_history = init_history(soltype, FT)
-    y_history = init_history(soltype, FT)
-    if soltype isa VerboseSolution
-        y, y′ = value_deriv(f, x0)
-        push_history!(x_history, x0, soltype)
-        push_history!(y_history, y, soltype)
-    end
-    for i in 1:maxiters
-        y, y′ = value_deriv(f, x0)
-        x1 = x0 - y / y′
-        push_history!(x_history, x1, soltype)
-        push_history!(y_history, y, soltype)
-        if tol(x0, x1, y)
-            return SolutionResults(
-                soltype,
-                x1,
-                true,
-                y,
-                i,
-                x_history,
-                y_history,
-            )
-        end
-        x0 = x1
-    end
-    return SolutionResults(
-        soltype,
-        x0,
-        false,
-        y,
-        maxiters,
-        x_history,
-        y_history,
-    )
+    return _find_zero_newton((x) -> value_deriv(f, x), f, x0, soltype, tol, maxiters)
```

Comment:
I don't _think_ this should matter too much, but I would be inclined to write this as
```suggestion
   return _find_zero_newton(Base.Fix1(value_deriv, f), f, x0, soltype, tol, maxiters)
```
which might help with type stability in some settings.